### PR TITLE
Add POC telemetry handler

### PR DESCRIPTION
**Pull request by nambrot**
### Description

Adds a POC for handling telemetry in contractkit

### CODE REVIEW

File: packages/contractkit/src/utils/rpc-caller.ts [hunk 1]
```diff
@@ -110,3 +128,13 @@ export class DefaultRpcCaller implements RpcCaller {
...
   }
 }
+
+export interface TelemetryHandler {
+  handleEvent: (path: string[], measurements: { duration: number }, metadata: any) => void
+}
+
+export class DebugTelemetryHandler implements TelemetryHandler {
+  handleEvent(path: string[], measurements: { duration: number }, metadata: any) {
+    debugRpcTelemetry('Path: %o, Duration: %s, metadata: %o', path, measurements.duration, metadata)
```
Comment (mcortesi) [inline packages/contractkit/src/utils/rpc-caller.ts:151]: where do you want to use this? You're using the debug handler, which is not a great fit if this is something you want always "on" for some cases

File: packages/contractkit/src/utils/rpc-caller.ts [hunk 2]
```diff
@@ -110,3 +128,13 @@ export class DefaultRpcCaller implements RpcCaller {
     }
   }
 }
+
+export interface TelemetryHandler {
+  handleEvent: (path: string[], measurements: { duration: number }, metadata: any) => void
+}
+
+export class DebugTelemetryHandler implements TelemetryHandler {
+  handleEvent(path: string[], measurements: { duration: number }, metadata: any) {
```
Comment (mcortesi) [inline packages/contractkit/src/utils/rpc-caller.ts:150]: I would add some documentation here... 
What does this is supposed to mean... ? path, measurements? metadata? 

For me; it's something no one else would know what to do with this or what is for

**Review by mcortesi — APPROVED** [review]
approved w/comments

File: packages/contractkit/src/utils/rpc-caller.ts (hunk 1 above)
Comment (nambrot) [reply packages/contractkit/src/utils/rpc-caller.ts:151]: Yeah I think the debug handler is a misnomer as we will probably always want to just add it to `debug`? The intention was to use this in the mobile wallet to measure eth_calls

File: packages/contractkit/src/utils/rpc-caller.ts (hunk 2 above)
Comment (nambrot) [reply packages/contractkit/src/utils/rpc-caller.ts:150]: Thanks for the comments. This wasn't mean to be an actual PR but just as a way to start the conversation. Would certainly properly type and document the actual payload. Right now it's just to measure how long calls take

**Comment by aaronmgdr** [comment]
close?

**Comment by barbaraliau** [comment]
@nambrot is this PR still relevant?

**Comment by barbaraliau** [comment]
Closing as this PR is stale. Please reopen if it is still relevant!

**Pull request closed without merging**